### PR TITLE
fix recur bug

### LINKED ISSUE #126: recur command bug
usage: recur 1 
without r/prefix -> recurs without stating recurrence and assertion error when task has no recurrence

### PR DESCRIPTION
**Pull request by umergta**
fixes commands like these:

- recur (no index, no recurrence)
- recur 1 (no r/)
- recur 1 r/ (no valid recurrence)

For now, recur r/ does not remove the recurrence but it can be done if yall want. (I think this should be done).
Secondly, if the only way for you to remove recurrence should still be through edit, maybe the name of this command should be changed to _addrecur_ instead. What do you guys think?

EDIT: 
- recur r/ : will remove recurrence

please let me know if you see any other bugs.

### CODE REVIEW

**Review by yhtMinceraft1010X — COMMENTED** [review]
LGTM, I think we should use `recur 1 r/` to remove recurrence

**Review by AiwassPrime — COMMENTED** [review]
These changes look good to me.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -273,20 +273,22 @@ Examples:
...
 
 Format: `recur INDEX r/RECURRENCE`
 
 * Recurs the task at the specified `INDEX`.
 * `INDEX` refers to the index number displayed in ModuleBook3.5.
    It must be a positive integer.
 * `RECURRENCE` refers to the regularity of the task that recurs periodically.
-* `RECURRENCE` can only be `daily`, `weekly` or `monthly`. It is case-sensitive.
+* `RECURRENCE` can only be `daily`, `weekly` or `monthly` if a task needs to be recurred.
+* `RECURRENCE` should be left empty if the recurrence of a task needs to be removed.
```
Comment (figo2127) [inline docs/UserGuide.md:299]: I thought this implementation is quite nice. Good work!

File: src/main/java/seedu/module/logic/commands/RecurCommand.java [hunk 2]
```diff
@@ -59,27 +64,37 @@ public CommandResult execute(Model model) throws CommandException {
...
+        }
 
-    private Task makeNextRecurringTask(Task previousRecurringTask) throws CommandException {
-        assert previousRecurringTask != null;
+        return new CommandResult(returnMessage);
+    }
 
-        OptionalField<Recurrence> recurrenceWrapper = new OptionalField<>(recurrence);
+    private void emptyRecurrenceHandler(OptionalField<Recurrence> recurrenceOptionalField) throws CommandException {
+        assert recurrenceOptionalField != null;
```
Comment (figo2127) [inline src/main/java/seedu/module/logic/commands/RecurCommand.java:88]: like the defensive code!

File: src/main/java/seedu/module/model/ModuleBook.java [hunk 3]
```diff
@@ -99,7 +98,7 @@ public void addTask(Task p) {
             assert(ModuleManager.moduleIsValid(p.getModule().toString()));
 
             Task taskToAdd = p;
-            //check the deadline and recurrence of the task
+
```
Comment (figo2127) [inline src/main/java/seedu/module/model/ModuleBook.java:101]: maybe its better to keep useful comments?

**Review by figo2127 — COMMENTED** [review]
Overall LGTM!

File: docs/UserGuide.md (hunk 1 above)
Comment (yhtMinceraft1010X) [reply docs/UserGuide.md:299]: Looks good, although I think the line for empty recurrence should also mention that the prefix `r/` must be kept.

**Review by QY-H00 — COMMENTED** [review]
Overall LGTM. Nice done!

File: docs/UserGuide.md (hunk 1 above)
Comment (umergta) [reply docs/UserGuide.md:299]: resolved

**Pull request merged**